holoviews/core/data/grid.py

```diff
@@ -231,6 +231,8 @@ def key_select_mask(cls, dataset, values, ind):
                 mask &= ind.start <= values
             if ind.stop is not None:
                 mask &= values < ind.stop
+            if mask is True:
+                mask = np.ones(values.shape, dtype=np.bool)
```

Comment:
Looks like ``mask`` is overloaded here - it seems to be both the boolean predicate figuring out if you want to create a mask array and the mask itself. I think it could be less confusing...